get_goes_x.py

```diff
@@ -31,44 +41,63 @@
     print(('Warning: failed to open URL {}: {}'.format(url, last_err)))
     sys.exit(0)
 
-
-colnames = 'year month dom  hhmm  mjd secs short long ratio'.split()
-data_colnames = colnames[-3:]
-
 try:
-    dat = ascii.read(urldat, guess=False, Reader=ascii.NoHeader,
-                     data_start=3, names=colnames)
+    dat = Table(json.loads(urldat))
 except Exception as err:
     print(('Warning: malformed GOES_X data so table read failed: {}'
           .format(err)))
     sys.exit(0)
 
-# Strip up to two rows at the end if any values are bad (i.e. negative)
-for _ in range(2):
-    if any(dat[name][-1] < 0 for name in data_colnames):
-        dat = dat[:-1]
+# Select only the GOES satellite specified from the args
+dat = dat[dat['satellite'] == args.satellite]
+if len(dat) == 0:
+    print('Warning: No data in fetched file for satellite {}'.format(args.satellite))
+    sys.exit(0)
 
-mjd = dat['mjd'] + dat['secs'] / 86400.
+# Make a table for each of the two wavelengths
+shortdat = dat[dat['energy'] == '0.05-0.4nm']['flux', 'time_tag']
+shortdat.rename_column('flux', 'short')
+longdat = dat[dat['energy'] == '0.1-0.8nm']['flux', 'time_tag']
+longdat.rename_column('flux', 'long')
+if len(longdat) != len(shortdat):
+    print('Warning: "short" and "long" table have mismatched lengths')
 
-secs = DateTime(mjd, format='mjd').secs
+# Join them on time (seems to be OK for these data)
+joindat = join(shortdat, longdat)
 
-descrs = dat.dtype.descr
-descrs.append(('time', 'f8'))
-newdat = np.ndarray(len(dat), dtype=descrs)
-for colname in colnames:
-    newdat[colname] = dat[colname]
-newdat['time'] = secs
+# Add a time column with Chandra secs and remove original time_tags
+times = Time(joindat['time_tag'])
+joindat['time'] = times.cxcsec
+joindat.remove_column('time_tag')
+
+# Add the other columns the old file format wanted
```

Comment:
I don't know if I should have just left these with placeholder values, but it seemed more work in the end to figure out what numbers to use if I didn't just put in values.